test/exportsField.js

```diff
@@ -0,0 +1,739 @@
+			expect: new Error(),
+			suite: [
+				{
+					"./utils/": "./../src/"
+				},
+				"./utils/index",
+				[]
+			]
+		},
+		{
+			name: "backtracking package base #5",
+			// fits package base depth
+			expect: ["./src/../index.js"],
+			suite: [
+				{
+					"./utils/index": "./src/../index.js"
+				},
+				"./utils/index",
+				[]
+			]
+		},
+		{
+			name: "backtracking package base #6",
+			// fits package base depth,
+			// but matching works as string concatenation not file paths
+			expect: [],
+			suite: [
+				{
+					"./utils/../utils/index": "./src/../index.js"
+				},
+				"./utils/index",
+				[]
+			]
+		},
+		{
+			name: "backtracking package base #7",
+			// enhanced-resolve don't know is this same package or not
+			expect: new Error(),
+			suite: [
+				{
+					"./utils/": {
+						browser: "../this/"
+					}
+				},
+				"./utils/index",
+				["browser"]
+			]
+		},
+		{
+			name: "backtracking package base #8",
+			expect: ["./utils/../index"],
+			suite: [
+				{
+					"./utils/": {
+						browser: "./utils/../"
+					}
+				},
+				"./utils/index",
+				["browser"]
+			]
+		},
+		//#endregion
+
+		//#region Directory exports subpaths may not backtrack above the target folder
+		{
+			name: "backtracking target folder #1",
+			expect: new Error(),
+			suite: [
+				{
+					"./utils/": "./dist/"
+				},
+				"./utils/timezone/../../index",
+				[]
+			]
+		},
+		{
+			name: "backtracking target folder #2",
+			expect: ["./dist/timezone/../index"],
+			suite: [
+				{
+					"./utils/": "./dist/"
+				},
+				"./utils/timezone/../index",
+				[]
+			]
+		},
+		{
+			name: "backtracking target folder #3",
+			expect: new Error(),
+			suite: [
+				{
+					"./utils/": "./dist/target/"
+				},
+				"./utils/../../index",
+				[]
+			]
+		},
+		//#endregion
+
+		//#region Exports targets cannot map into a nested node_modules path
+		{
+			name: "nested node_modules path #1",
+			expect: new Error(),
+			suite: [
+				{
+					"./utils/": {
+						browser: "./node_modules/"
+					}
+				},
+				"./utils/lodash/dist/index.js",
+				["browser"]
+			]
+		},
+		{
+			name: "nested node_modules path #2",
+			expect: new Error(),
+			suite: [
+				{
+					"./utils/": "./utils/../node_modules/"
+				},
+				"./utils/lodash/dist/index.js",
+				[]
+			]
+		},
+		{
+			name: "nested node_modules path #3",
+			expect: new Error(),
+			suite: [
+				{
+					"./utils/index": "./utils/../node_modules/index"
+				},
+				"./utils/index",
+				[]
+			]
+		},
+		{
+			name: "nested node_modules path #4",
+			expect: new Error(),
+			suite: [
+				{
+					"./utils/index": "./node_modules/index"
+				},
+				"./utils/index",
+				[]
+			]
+		}
+		//#endregion
+	];
+
+	testCases.forEach(testCase => {
+		it(testCase.name, () => {
+			if (testCase.expect instanceof Error) {
+				should.throws(() => processExportsField(...testCase.suite));
+			} else {
+				processExportsField(...testCase.suite).should.containDeepOrdered(
```

Comment:
Why only contains and not eql?

Reply:
Don't much familiar with `should.js`. fixed